Extensions/PathfindingBehavior/tests/legacypathfindingruntimebehavior.spec.js

```diff
@@ -0,0 +1,689 @@
+        bottomObstacleY =
+          playerY + playerBottomBorder + obstacleTopBorder + inclusiveMargin;
+        targetX =
+          obstacleX + obstacleRightBorder + playerLeftBorder + 2 * cellSize;
+        targetY = playerY;
+        straitLineCells = (targetX - playerX) / cellSize + 1;
+        break;
+      }
+      case 'left': {
+        const obstacleX =
+          playerX - playerLeftBorder - 2 * cellSize - obstacleRightBorder;
+        topObstacleX = obstacleX;
+        bottomObstacleX = obstacleX;
+        topObstacleY =
+          playerY - playerTopBorder - obstacleBottomBorder - inclusiveMargin;
+        bottomObstacleY =
+          playerY + playerBottomBorder + obstacleTopBorder + inclusiveMargin;
+        targetX =
+          obstacleX - obstacleLeftBorder - playerRightBorder - 2 * cellSize;
+        targetY = playerY;
+        straitLineCells = (playerX - targetX) / cellSize + 1;
+        break;
+      }
+      case 'up': {
+        // topObstacle and bottomObstacle are actually left and right but for convegnance...
+        const obstacleY =
+          playerY - playerTopBorder - 2 * cellSize - obstacleBottomBorder;
+        topObstacleY = obstacleY;
+        bottomObstacleY = obstacleY;
+        topObstacleX =
+          playerX - playerLeftBorder - obstacleRightBorder - inclusiveMargin;
+        bottomObstacleX =
+          playerX + playerRightBorder + obstacleLeftBorder + inclusiveMargin;
+        targetY =
+          obstacleY - obstacleTopBorder - playerBottomBorder - 2 * cellSize;
+        targetX = playerX;
+        straitLineCells = (playerY - targetY) / cellSize + 1;
+        break;
+      }
+      case 'down': {
+        // topObstacle and bottomObstacle are actually left and right but for convegnance...
+        const obstacleY =
+          playerY + playerBottomBorder + 2 * cellSize + obstacleTopBorder;
+        topObstacleY = obstacleY;
+        bottomObstacleY = obstacleY;
+        topObstacleX =
+          playerX - playerLeftBorder - obstacleRightBorder - inclusiveMargin;
+        bottomObstacleX =
+          playerX + playerRightBorder + obstacleLeftBorder + inclusiveMargin;
+        targetY =
+          obstacleY + obstacleBottomBorder + playerTopBorder + 2 * cellSize;
+        targetX = playerX;
+        straitLineCells = (targetY - playerY) / cellSize + 1;
+        break;
+      }
+    }
+
+    obstacleTop.setPosition(topObstacleX, topObstacleY);
+    obstacleBottom.setPosition(bottomObstacleX, bottomObstacleY);
+    // so obstacles register
+    runtimeScene.renderAndStep(1000 / 60);
+
+    player.setPosition(playerX, playerY);
+    player.getBehavior(pathFindingName).moveTo(runtimeScene, targetX, targetY);
+    expect(player.getBehavior(pathFindingName).pathFound()).to.be(true);
+    expect(player.getBehavior(pathFindingName).getNodeCount()).to.be(
+      straitLineCells
+    );
+  });
+
+  it("mustn't find a direct path between 2 slightly too much tighten obstacles", function () {
```

Comment:
"mustn't find a direct path between 2 slightly too much tighten obstacles"
=>
"should not find a direct path between 2 obstacles making a path slightly too narrow."